src/System.Xml.XDocument/System/Xml/Linq/XElement.cs

```diff
@@ -136,11 +138,19 @@ internal XElement(XmlReader r)
         {
         }
 
+        internal XElement(AsyncConstructionSentry s)
```

Comment:
The sync counterpart XElement(XmlReader r, LoadOptions o) bypasses initialization that the default constructor does.

Some sort of overloaded constructor is needed to maintain the same behavior, but beyond simply bypassing that initialization, everything else that needs to be done happens after some I/O.